spec/lib/resque-serializer/monkey_patches/resque_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'spec_helper'
+
+class ResqueDequeueHookJob
+  extend Resque::Plugins::Serializer
+
+  serialize :both
+
+  @queue = :default
+
+  def self.perform(*args); end
+end
+
+RSpec.describe ResqueDequeueHookJob do
+  let(:args) { %w[arg1 arg2] }
+  let(:dequeue_hooks) { [] }
+
+  before do
+    allow(Resque::Plugin)
+      .to receive(:before_dequeue_hooks)
+      .and_return(dequeue_hooks)
+    allow(Resque::Plugin)
+      .to receive(:after_dequeue_hooks)
+      .and_return(dequeue_hooks)
+    Resque.push(
+      :default,
+      class: described_class.to_s,
+      args: args
+    )
+    Resque.pop(:default)
+  end
+
+  specify do
+    expect(Resque::Plugin)
+      .to have_received(:before_dequeue_hooks)
+      .ordered
+    expect(Resque::Plugin)
+      .to have_received(:after_dequeue_hooks)
+      .ordered
+  end
```

Comment:
Let's rewrite this to explicitly call the subject; it makes the tests a lot less brittle.
```rb
before do
  Resque.push(:default, class: described_class.to_s, args: args)
end

subject(:pop) { Resque.pop(:default) }

it 'executes the dequeue hooks' do
  expect(Resque::Plugin).to receive(:before_dequeue_hooks).ordered
  expect(Resque::Plugin).to receive(:before_dequeue_hooks).ordered

  pop
end
```